File: DSA/DSA-C/ARRAY/sum_subarray.c
// Maximum sum subrarry problem
// Brute Force method
#include<stdio.h>
#include<math.h>
int main(){
    int i, j;
    int size = 7; int w = 4; // w is the size of subarray
    int arr[7] ={3,8,2,5,7,6,12};
   float maxx = -INFINITY;
   int current;

    for(i = 0; i<=(size-w); i++){

        current = 0;

        for ( j = i; j<=(i+w-1); j++)
        {
            current = current + arr[j];

            if(current >= maxx){
                maxx = current;
                printf("Maxx = %f\n", maxx);
            }
            else{
                maxx = maxx;
                printf("Maxx = %f\n", maxx);
            }
            
        }
      
    }
    printf("So the max value of sum of subarray is :%f",maxx); 
}

File: DSA/DSA-PYTHON/ARRAY/sum_subarray.py
arr = [3,8,2,5,7,6,12]
w = 4
maxx = -float('inf')
for i in range(len(arr)-w+1):
    current = 0
    for j in range(i , i+w):
        current = current + arr[j]
        maxx = max(current,maxx)

print(f"So the sum of subarray is {maxx}")
    